modules/docs/arrow-docs/docs/docs/arrow/data/reader/README.md

```diff
@@ -11,6 +11,214 @@ redirect_from:
 {:.intermediate}
 intermediate
 
+Reader is a datatype that represents computation that can inject values from a shared environment.
```

Comment:
Perhaps I'd reformulate to also have the verb `read` appear (that would give a better hint at why it's called like `reader` in the first place).

Perhaps something like `Reader is a datatype representing computations that read values from a shared environment.`

Although, I understand why you used `inject`, as it can be seen as a form of _dependency injection_ mechanism.

What do you think?